_posts/2022-10-28-testing-primer-dev-academy.md

```diff
@@ -0,0 +1,101 @@
+---
+layout: post
+title: Testing primer - lessons to learn from Solita development academy exercises
+author: hjhamala
+excerpt: >
+  
+tags:
+  - testing
+  - career development
+  - coding
+---
+The Solita developer [academy](https://www.solita.fi/en/academy/) is a way to kickstart a career in software development. Applying for the academy fall 2022 required to make [a pre-assignemt](https://github.com/solita/dev-academy-2022-fall-exercise). The assignment was based on the Helsinki Capital area city bike journey statistics. We received many great exercises.  
+
+One surprising thing was that there were few or no tests in many applications. Pre-assignment mentioned tests quite clearly. After going through the exercises I found some of the exercises mentioned that an applicant had found difficult to make meaningful tests. 
+ 
+I think that as professional developers we may have forgotten that writing good and useful tests have not been an easy thing to learn - not for me at least. Therefore I present a quick primer based on Solita's fall assignment and how it could be tested. The blog post is mainly aimed at beginner developers. 
+
+## Why do we test?
+First, we should think about the reason why we even bother to write tests. The next three reasons rise to my mind
+* Tests are quality control. Tests tell us that the things we are developing do the right things - the right way.
+* Tests help find out regressions. Changing the code like refactoring is easier and less nervous when we have confidence that the software works after the changes.
+* Tests are part of the documentation. In many cases, tests can be used to find out how libraries/APIs are expected to be used.
+
+## How we test
+Classically tests are split into three main types of tests.
+* Unit tests test software modules in isolation from each other. 
+* Integration tests are used to find out that components/modules work together
+* E2E tests test the whole application - for a web application this could mean testing the application flow with an automated browser
+For small applications - like Solita assignment - it may be quite hard to make a clear difference between integration and unit especially if the application is not split into clear components.
```

Comment:
Here it seems a  period "." is missing?

Reply:
Seems to miss a line break. "For small application" should be its own paragraph